<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Changelog • dasper</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Changelog"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-news">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">dasper</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.5.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/dasper.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/dzhang32/dasper/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
      <h1 data-toc-skip>Changelog <small></small></h1>
      <small>Source: <a href="https://github.com/dzhang32/dasper/blob/HEAD/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
    </div>

    <div class="section level2">
<h2 class="page-header" data-toc-text="1.5.1" id="dasper-151">dasper 1.5.1<a class="anchor" aria-label="anchor" href="#dasper-151"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Fix bug related to <code><a href="../reference/coverage_process.html">coverage_norm()</a></code>, ensuring that
regions used to normalise coverage have the same <code>seqlevels</code>
as the inputted <code>junctions</code>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.3.8" id="dasper-138">dasper 1.3.8<a class="anchor" aria-label="anchor" href="#dasper-138"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Fix bug related to <code><a href="https://pkgdown.r-lib.org/reference/deploy_to_branch.html" class="external-link">pkgdown::deploy_to_branch()</a></code>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.3.6" id="dasper-136">dasper 1.3.6<a class="anchor" aria-label="anchor" href="#dasper-136"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Add functionality to annotate junctions with
<code>gene_name</code>/<code>symbol</code> using <code>EnsDb</code>
inputted into <code><a href="../reference/junction_process.html">junction_annot()</a></code>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.3.2" id="dasper-132">dasper 1.3.2<a class="anchor" aria-label="anchor" href="#dasper-132"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Fix bugs within <code><a href="../reference/plot_sashimi.html">plot_sashimi()</a></code> and enable the
visualization of raw junction counts.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="1.1.1" id="dasper-111">dasper 1.1.1<a class="anchor" aria-label="anchor" href="#dasper-111"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Use of testthat edition 3 and parrallel running of tests.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.2" id="dasper-0992">dasper 0.99.2<a class="anchor" aria-label="anchor" href="#dasper-0992"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Merge documentation into one man page for junction, coverage and
outlier processing functions to reduce runtime of roxygen examples.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.1" id="dasper-0991">dasper 0.99.1<a class="anchor" aria-label="anchor" href="#dasper-0991"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Change <code><a href="../reference/outlier_process.html">outlier_detect()</a></code> to using <code>basilisk</code>
for interfacing into python replacing <code>reticulate</code>.</li>
</ul></div>
    <div class="section level2">
<h2 class="page-header" data-toc-text="0.99.0" id="dasper-0990">dasper 0.99.0<a class="anchor" aria-label="anchor" href="#dasper-0990"></a></h2>
<p>NEW FEATURES</p>
<ul><li>Converted <code>dasper</code> into a Bioconductor-friendly format
using <a href="https://lcolladotor.github.io/biocthis/" class="external-link">biocthis</a>.</li>
<li>Added <code><a href="../reference/junction_load.html">junction_load()</a></code>, which loads raw junction data
from RNA-sequencing into an <code>RangedSummarizedExperiment</code>
object. Includes an option to allow download of user-specified control
junctions.</li>
<li>Added <code><a href="../reference/junction_process.html">junction_annot()</a></code>, which uses information from
reference annotation and the strand of a junction to classify junctions
as “annotated”, “novel_acceptor”, “novel_donor”, “novel_exon_skip”,
“novel_combo”, “ambig_gene” and “unannotated”.</li>
<li>Added <code><a href="../reference/junction_process.html">junction_filter()</a></code>, which filters junctions by
their count, width, annotation or if they overlap a set of user-defined
regions.</li>
<li>Added <code><a href="../reference/junction_process.html">junction_norm()</a></code>, which normalises raw junction
counts (into a proportion-spliced-in) by dividing the counts of each
junction by the total number of counts in it’s associated cluster.</li>
<li>Added <code><a href="../reference/junction_process.html">junction_process()</a></code>, a wrapper function for all
“junction_” prefixed functions except <code><a href="../reference/junction_load.html">junction_load()</a></code>.</li>
<li>Added <code><a href="../reference/junction_process.html">junction_score()</a></code>, which scores patient junctions
based on the extent their counts deviate from a control count
distribution.</li>
<li>Added <code><a href="../reference/coverage_process.html">coverage_norm()</a></code>, which will load and normalise
coverage for exonic/intronic regions corresponding to each
junction.</li>
<li>Added <code><a href="../reference/coverage_process.html">coverage_score()</a></code>, which scores coverage
associated with each junction based on it’s deviation from control
coverage distributions.</li>
<li>Added <code><a href="../reference/coverage_process.html">coverage_process()</a></code>, a wrapper function for all
“coverage_” prefixed functions.</li>
<li>Added <code><a href="../reference/outlier_process.html">outlier_detect()</a></code>, which uses the junction scores
and coverage scores as input into an unsupervised outlier detection
algorithm to find the most outlier-looking junctions in each
sample.</li>
<li>Added <code><a href="../reference/outlier_process.html">outlier_aggregate()</a></code>, which aggregates the
junction-level outlier data to a cluster-level.</li>
<li>Added <code><a href="../reference/outlier_process.html">outlier_process()</a></code>, a wrapper function for all
“outlier_” prefixed functions.</li>
<li>Added <code><a href="../reference/plot_sashimi.html">plot_sashimi()</a></code>, which enables the visualisation
of junction data across genes/transcripts or regions of interest.</li>
</ul></div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>

</div>


      <footer><div class="copyright">
  <p></p><p>Developed by David Zhang.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

