<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Processing coverage — coverage_norm • dasper</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Processing coverage — coverage_norm"><meta property="og:description" content='The set of functions prefixed with "coverage_" are used to
process coverage data. They are designed to be run after you have processed
your junctions in the order coverage_norm, coverage_score. Or,
alternatively the wrapper function coverage_process can be used to run
the 2 functions stated above in one go. For more details of the individual
functions, see "Details".'><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">dasper</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.5.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/dasper.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/dzhang32/dasper/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Processing coverage</h1>
    <small class="dont-index">Source: <a href="https://github.com/dzhang32/dasper/blob/HEAD/R/coverage_norm.R" class="external-link"><code>R/coverage_norm.R</code></a>, <a href="https://github.com/dzhang32/dasper/blob/HEAD/R/coverage_process.R" class="external-link"><code>R/coverage_process.R</code></a>, <a href="https://github.com/dzhang32/dasper/blob/HEAD/R/coverage_score.R" class="external-link"><code>R/coverage_score.R</code></a></small>
    <div class="hidden name"><code>coverage_process.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The set of functions prefixed with "coverage_" are used to
process coverage data. They are designed to be run after you have processed
your junctions in the order <code>coverage_norm</code>, <code>coverage_score</code>. Or,
alternatively the wrapper function <code>coverage_process</code> can be used to run
the 2 functions stated above in one go. For more details of the individual
functions, see "Details".</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">coverage_norm</span><span class="op">(</span>
  <span class="va">junctions</span>,
  <span class="va">ref</span>,
  unannot_width <span class="op">=</span> <span class="fl">20</span>,
  <span class="va">coverage_paths_case</span>,
  <span class="va">coverage_paths_control</span>,
  coverage_chr_control <span class="op">=</span> <span class="cn">NULL</span>,
  load_func <span class="op">=</span> <span class="va">.coverage_load</span>,
  bp_param <span class="op">=</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/SerialParam-class.html" class="external-link">SerialParam</a></span><span class="op">(</span><span class="op">)</span>,
  norm_const <span class="op">=</span> <span class="fl">1</span>
<span class="op">)</span>

<span class="fu">coverage_process</span><span class="op">(</span>
  <span class="va">junctions</span>,
  <span class="va">ref</span>,
  unannot_width <span class="op">=</span> <span class="fl">20</span>,
  <span class="va">coverage_paths_case</span>,
  <span class="va">coverage_paths_control</span>,
  coverage_chr_control <span class="op">=</span> <span class="cn">NULL</span>,
  load_func <span class="op">=</span> <span class="va">.coverage_load</span>,
  bp_param <span class="op">=</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/SerialParam-class.html" class="external-link">SerialParam</a></span><span class="op">(</span><span class="op">)</span>,
  norm_const <span class="op">=</span> <span class="fl">1</span>,
  score_func <span class="op">=</span> <span class="va">.zscore</span>,
  <span class="va">...</span>
<span class="op">)</span>

<span class="fu">coverage_score</span><span class="op">(</span><span class="va">junctions</span>, <span class="va">coverage</span>, score_func <span class="op">=</span> <span class="va">.zscore</span>, <span class="va">...</span><span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>junctions</dt>
<dd><p>junction data as a
<a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html" class="external-link">RangedSummarizedExperiment-class</a>
object.</p></dd>
<dt>ref</dt>
<dd><p>either path to gtf/gff3 or object of class
<a href="https://rdrr.io/pkg/GenomicFeatures/man/TxDb-class.html" class="external-link">TxDb-class</a> or
<a href="https://rdrr.io/pkg/ensembldb/man/EnsDb-class.html" class="external-link">EnsDb-class</a>.
<a href="https://rdrr.io/pkg/ensembldb/man/EnsDb-class.html" class="external-link">EnsDb-class</a> is required if you intend to annotate
junctions with gene symbols/names.</p></dd>
<dt>unannot_width</dt>
<dd><p>integer scalar determining the width of the region to
obtain coverage from when the end of of a junction does not overlap an
existing exon.</p></dd>
<dt>coverage_paths_case</dt>
<dd><p>paths to the BigWig files containing the coverage
of your case samples.</p></dd>
<dt>coverage_paths_control</dt>
<dd><p>paths to the BigWig files for control samples.</p></dd>
<dt>coverage_chr_control</dt>
<dd><p>either "chr" or "no_chr", indicating the
chromosome format of control coverage data. Only required if the
chromosome format of the control BigWig files is different to that of your
cases.</p></dd>
<dt>load_func</dt>
<dd><p>a function to use to load coverage. Currently only for
internal use to increase testing speed.</p></dd>
<dt>bp_param</dt>
<dd><p>a
<a href="https://rdrr.io/pkg/BiocParallel/man/BiocParallelParam-class.html" class="external-link">BiocParallelParam-class</a> instance
denoting whether to parallelise the loading of coverage across BigWig
files.</p></dd>
<dt>norm_const</dt>
<dd><p>numeric scaler to add to the normalisation coverage to
avoid dividing by 0s and resulting NaN or Inf values.</p></dd>
<dt>score_func</dt>
<dd><p>function to score junctions by their abnormality. By
default, will use a z-score but can be switched to a user-defined function.
This function must take as input an <code>x</code> and <code>y</code> argument, containing case
and control counts respectively. This must return a numeric vector equal to
the length of <code>x</code> with elements corresponding to a abnormality of each
junction.</p></dd>
<dt>...</dt>
<dd><p>additional arguments passed to <code>score_func</code>.</p></dd>
<dt>coverage</dt>
<dd><p>list containing normalised coverage data that is outputted
from coverage_norm.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>junctions as
<a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">SummarizedExperiment</a>object with additional <code>assays</code> named "coverage_region" and
"coverage_score". "coverage_region" labels the region of greatest
disruption (1 = exon_start, 2 = exon_end, 3 = intron) and "coverage_score"
contains the abnormality scores of the region with the greatest disruption.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p><code>coverage_process</code> wraps all "coverage_" prefixed functions in
<a href="dasper.html">dasper</a>. This is designed to simplify processing of the coverage data for
those familiar or uninterested with the intermediates.</p>
<p><code>coverage_norm</code> obtains regions of interest for each junction where
coverage disruptions would be expected. These consist of the intron itself
the overlapping exon definitions (if ends of junctions are annotated),
picking the shortest exon when multiple overlap one end. If ends are
unannotated, <code>coverage_norm</code> will use a user-defined width set by
<code>unannot_width</code>. Then, coverage will be loaded using
<a href="https://github.com/ChristopherWilks/megadepth" class="external-link">megadepth</a> and
normalised to a set region per junction.  By default, the boundaries of
each gene associated to a junction are used as the region to normalise to.</p>
<p><code>coverage_score</code> will score disruptions in the coverage across the
intronic/exonic regions associated with each junction. This abnormality
score generated by <code>score_func</code> operates by calculating the deviation of
the coverage in patients to a coverage across the same regions in controls.
Then, for each junction it obtains the score of the region with the
greatest disruption.</p>
    </div>
    <div id="functions">
    <h2>Functions</h2>
    
<ul><li><p><code>coverage_norm</code>: Load and normalise coverage from RNA-sequencing data</p></li>
<li><p><code>coverage_score</code>: Score coverage by their abnormality</p></li>
</ul></div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"></span>
<span class="r-in"><span class="co">##### Set up txdb #####</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># use GenomicState to load txdb (GENCODE v31)</span></span>
<span class="r-in"><span class="va">ref</span> <span class="op">&lt;-</span> <span class="fu">GenomicState</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicState/man/GenomicStateHub.html" class="external-link">GenomicStateHub</a></span><span class="op">(</span></span>
<span class="r-in">    version <span class="op">=</span> <span class="st">"31"</span>,</span>
<span class="r-in">    genome <span class="op">=</span> <span class="st">"hg38"</span>,</span>
<span class="r-in">    filetype <span class="op">=</span> <span class="st">"TxDb"</span></span>
<span class="r-in"><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> snapshotDate(): 2021-10-20</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomicFeatures</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: BiocGenerics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘BiocGenerics’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:stats’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     IQR, mad, sd, var, xtabs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Filter, Find, Map, Position, Reduce, anyDuplicated, append,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     as.data.frame, basename, cbind, colnames, dirname, do.call,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     tapply, union, unique, unsplit, which.max, which.min</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: S4Vectors</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: stats4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘S4Vectors’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     I, expand.grid, unname</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: IRanges</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomeInfoDb</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomicRanges</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: AnnotationDbi</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: Biobase</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Welcome to Bioconductor</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Vignettes contain introductory material; view with</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     'browseVignettes()'. To cite Bioconductor, see</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     'citation("Biobase")', and for packages 'citation("pkgname")'.</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">##### Set up BigWig #####</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># obtain path to example bw on recount2</span></span>
<span class="r-in"><span class="va">bw_path</span> <span class="op">&lt;-</span> <span class="fu">recount</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/recount/man/download_study.html" class="external-link">download_study</a></span><span class="op">(</span></span>
<span class="r-in">    project <span class="op">=</span> <span class="st">"SRP012682"</span>,</span>
<span class="r-in">    type <span class="op">=</span> <span class="st">"samples"</span>,</span>
<span class="r-in">    download <span class="op">=</span> <span class="cn">FALSE</span></span>
<span class="r-in"><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Setting options('download.file.method.GEOquery'='auto')</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Setting options('GEOquery.inmemory.gpl'=FALSE)</span>
<span class="r-in"><span class="co"># \dontshow{</span></span>
<span class="r-in"><span class="co"># cache the bw for speed in later</span></span>
<span class="r-in"><span class="co"># examples/testing during R CMD Check</span></span>
<span class="r-in"><span class="va">bw_path</span> <span class="op">&lt;-</span> <span class="fu">dasper</span><span class="fu">:::</span><span class="fu">.file_cache</span><span class="op">(</span><span class="va">bw_path</span><span class="op">)</span></span>
<span class="r-in"><span class="co"># }</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">##### junction_process #####</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">junctions_processed</span> <span class="op">&lt;-</span> <span class="fu"><a href="junction_process.html">junction_process</a></span><span class="op">(</span></span>
<span class="r-in">    <span class="va">junctions_example</span>,</span>
<span class="r-in">    <span class="va">ref</span>,</span>
<span class="r-in">    types <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ambig_gene"</span>, <span class="st">"unannotated"</span><span class="op">)</span>,</span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "# Filtering junctions -----------------------------------------------------"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:13 - Filtering junctions..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:13 - by count..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:13 - done!"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "# Annotating junctions ----------------------------------------------------"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:13 - Obtaining co-ordinates of annotated exons and junctions..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:33 - Getting junction annotation using overlapping exons..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:34 - Tidying junction annotation..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:34 - Deriving junction categories..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:35 - done!"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "# Filtering junctions -----------------------------------------------------"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:35 - Filtering junctions..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:35 - by type..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:35 - done!"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "# Normalise junctions -----------------------------------------------------"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:35 - Clustering junctions..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:35 - Normalising junction counts..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:35 - done!"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "# Score junctions ---------------------------------------------------------"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:35 - Calculating the direction of change of junctions..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:35 - Generating junction abnormality score..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:36 - done!"</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">##### install megadepth #####</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># required to load coverage in coverage_norm()</span></span>
<span class="r-in"><span class="fu">megadepth</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/megadepth/man/install_megadepth.html" class="external-link">install_megadepth</a></span><span class="op">(</span>force <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The latest megadepth version is 1.2.0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> This is not an interactive session, therefore megadepth has been installed temporarily to </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> /tmp/RtmpMrdPeC/megadepth</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">##### coverage_norm #####</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">coverage_normed</span> <span class="op">&lt;-</span> <span class="fu">coverage_norm</span><span class="op">(</span></span>
<span class="r-in">    <span class="va">junctions_processed</span>,</span>
<span class="r-in">    <span class="va">ref</span>,</span>
<span class="r-in">    unannot_width <span class="op">=</span> <span class="fl">20</span>,</span>
<span class="r-in">    coverage_paths_case <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="va">bw_path</span>, <span class="fl">2</span><span class="op">)</span>,</span>
<span class="r-in">    coverage_paths_control <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="va">bw_path</span>, <span class="fl">2</span><span class="op">)</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:37 - Obtaining exonic and intronic regions to load coverage from..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:38 - Obtaining regions to use to normalise coverage..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:43 - Loading coverage..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:53 - Normalising coverage..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:53 - done!"</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">##### coverage_score #####</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">junctions</span> <span class="op">&lt;-</span> <span class="fu">coverage_score</span><span class="op">(</span><span class="va">junctions_processed</span>, <span class="va">coverage_normed</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:53 - Generating coverage abnormality score..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:54 - Obtaining regions with greatest coverage dysruption..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:54 - done!"</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">##### coverage_process #####</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># this wrapper will obtain coverage scores identical to those</span></span>
<span class="r-in"><span class="co"># obtained through running the individual wrapped functions shown below</span></span>
<span class="r-in"><span class="va">junctions_w_coverage</span> <span class="op">&lt;-</span> <span class="fu">coverage_process</span><span class="op">(</span></span>
<span class="r-in">    <span class="va">junctions_processed</span>,</span>
<span class="r-in">    <span class="va">ref</span>,</span>
<span class="r-in">    coverage_paths_case <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="va">bw_path</span>, <span class="fl">2</span><span class="op">)</span>,</span>
<span class="r-in">    coverage_paths_control <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="va">bw_path</span>, <span class="fl">3</span><span class="op">)</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "# Loading and normalising coverage ---------------------------------------------"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:54 - Obtaining exonic and intronic regions to load coverage from..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:58:54 - Obtaining regions to use to normalise coverage..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:59:00 - Loading coverage..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:59:12 - Normalising coverage..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:59:12 - done!"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "# Scoring coverage ---------------------------------------------"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:59:12 - Generating coverage abnormality score..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:59:13 - Obtaining regions with greatest coverage dysruption..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-26 18:59:13 - done!"</span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># the two objects are equivalent</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/all.equal.html" class="external-link">all.equal</a></span><span class="op">(</span><span class="va">junctions_w_coverage</span>, <span class="va">junctions</span>, check.attributes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] TRUE</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by David Zhang.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

