<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Load junctions from RNA-sequencing data — junction_load • dasper</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Load junctions from RNA-sequencing data — junction_load"><meta property="og:description" content="junction_load loads in raw patient and control junction data and formats it
into a
RangedSummarizedExperiment-class
object. Control samples can be the user's in-house samples or selected from
GTEx v6 data publicly released through the
recount2 and
downloaded through snaptron. By default,
junction_load expects the junction data to be in STAR aligned format
(SJ.out) but this can be modified via the argument load_func."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">dasper</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.5.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/dasper.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/dzhang32/dasper/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Load junctions from RNA-sequencing data</h1>
    <small class="dont-index">Source: <a href="https://github.com/dzhang32/dasper/blob/HEAD/R/junction_load.R" class="external-link"><code>R/junction_load.R</code></a></small>
    <div class="hidden name"><code>junction_load.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>junction_load</code> loads in raw patient and control junction data and formats it
into a
<a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html" class="external-link">RangedSummarizedExperiment-class</a>
object. Control samples can be the user's in-house samples or selected from
GTEx v6 data publicly released through the
<a href="https://jhubiostatistics.shinyapps.io/recount/" class="external-link">recount2</a> and
downloaded through <a href="http://snaptron.cs.jhu.edu/" class="external-link">snaptron</a>. By default,
<code>junction_load</code> expects the junction data to be in STAR aligned format
(SJ.out) but this can be modified via the argument <code>load_func</code>.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">junction_load</span><span class="op">(</span>
  <span class="va">junction_paths</span>,
  metadata <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span>samp_id <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_c.html" class="external-link">str_c</a></span><span class="op">(</span><span class="st">"samp_"</span>,
    <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">junction_paths</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,
  controls <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="cn">FALSE</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">junction_paths</span><span class="op">)</span><span class="op">)</span>,
  load_func <span class="op">=</span> <span class="va">.STAR_load</span>,
  chrs <span class="op">=</span> <span class="cn">NULL</span>,
  coord_system <span class="op">=</span> <span class="fl">1</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>junction_paths</dt>
<dd><p>path(s) to junction data.</p></dd>
<dt>metadata</dt>
<dd><p>data.frame containing sample metadata with rows in the same
order as <code>junction_paths</code>.</p></dd>
<dt>controls</dt>
<dd><p>either a logical vector of the same length as
<code>junction_paths</code> with TRUE representing controls. Or, one of "fibroblasts",
"lymphocytes", "skeletal_muscle", "whole_blood" representing the samples of
which GTEx tissue to use as controls. By default, will assume all samples
are patients.</p></dd>
<dt>load_func</dt>
<dd><p>function to load in junctions. By default, requires STAR
formatted junctions (SJ.out). But this can be switched dependent on the
format of the user's junction data. Function must take as input a junction
path then return a data.frame with the columns "chr", "start", "end",
"strand" and "count".</p></dd>
<dt>chrs</dt>
<dd><p>chromosomes to keep. By default, no filter is applied.</p></dd>
<dt>coord_system</dt>
<dd><p>1 (1-based) or 0 (0-based) denoting the co-ordinate
system corresponding to the user junctions from <code>junction_paths</code>. Only used
when controls is set to "fibroblasts" to ensure GTEx data is harmonised to
match the co-ordinate system of the user's junctions.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p><a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html" class="external-link">RangedSummarizedExperiment-class</a>object containing junction data.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"></span>
<span class="r-in"><span class="va">junctions_example_1_path</span> <span class="op">&lt;-</span></span>
<span class="r-in">    <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,</span>
<span class="r-in">        <span class="st">"junctions_example_1.txt"</span>,</span>
<span class="r-in">        package <span class="op">=</span> <span class="st">"dasper"</span>,</span>
<span class="r-in">        mustWork <span class="op">=</span> <span class="cn">TRUE</span></span>
<span class="r-in">    <span class="op">)</span></span>
<span class="r-in"><span class="va">junctions_example_2_path</span> <span class="op">&lt;-</span></span>
<span class="r-in">    <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,</span>
<span class="r-in">        <span class="st">"junctions_example_2.txt"</span>,</span>
<span class="r-in">        package <span class="op">=</span> <span class="st">"dasper"</span>,</span>
<span class="r-in">        mustWork <span class="op">=</span> <span class="cn">TRUE</span></span>
<span class="r-in">    <span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">junctions</span> <span class="op">&lt;-</span></span>
<span class="r-in">    <span class="fu">junction_load</span><span class="op">(</span></span>
<span class="r-in">        junction_paths <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">junctions_example_1_path</span>, <span class="va">junctions_example_2_path</span><span class="op">)</span></span>
<span class="r-in">    <span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:02:33 - Loading junctions for sample 1/2..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:02:33 - Loading junctions for sample 2/2..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:02:33 - Adding control junctions..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:02:33 - Tidying and storing junction data as a RangedSummarizedExperiment..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:02:33 - done!"</span>
<span class="r-in"></span>
<span class="r-in"><span class="va">junctions</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: RangedSummarizedExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 19733 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(1): raw</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(2): count_1 count_2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(2): samp_id case_control</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by David Zhang.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

