<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Processing junctions — junction_annot • dasper</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Processing junctions — junction_annot"><meta property="og:description" content='The set of functions prefixed with "junction_" are used to
process junction data. They are designed to be run in a sequential manner
in the order junction_annot, junction_filter, junction_norm,
junction_score. Or, alternatively the wrapper function junction_process
can be used to run all 4 of the functions stated above in one go. For more
details of the individual functions, see "Details".'><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">dasper</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.5.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/dasper.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/dzhang32/dasper/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Processing junctions</h1>
    <small class="dont-index">Source: <a href="https://github.com/dzhang32/dasper/blob/HEAD/R/junction_annot.R" class="external-link"><code>R/junction_annot.R</code></a>, <a href="https://github.com/dzhang32/dasper/blob/HEAD/R/junction_filter.R" class="external-link"><code>R/junction_filter.R</code></a>, <a href="https://github.com/dzhang32/dasper/blob/HEAD/R/junction_norm.R" class="external-link"><code>R/junction_norm.R</code></a>, and 2 more</small>
    <div class="hidden name"><code>junction_process.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The set of functions prefixed with "junction_" are used to
process junction data. They are designed to be run in a sequential manner
in the order <code>junction_annot</code>, <code>junction_filter</code>, <code>junction_norm</code>,
<code>junction_score</code>. Or, alternatively the wrapper function <code>junction_process</code>
can be used to run all 4 of the functions stated above in one go. For more
details of the individual functions, see "Details".</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">junction_annot</span><span class="op">(</span>
  <span class="va">junctions</span>,
  <span class="va">ref</span>,
  ref_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene_id"</span>, <span class="st">"tx_name"</span>, <span class="st">"exon_id"</span><span class="op">)</span>,
  ref_cols_to_merge <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene_id"</span><span class="op">)</span>
<span class="op">)</span>

<span class="fu">junction_filter</span><span class="op">(</span>
  <span class="va">junctions</span>,
  count_thresh <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>raw <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>,
  n_samp <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>raw <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>,
  width_range <span class="op">=</span> <span class="cn">NULL</span>,
  types <span class="op">=</span> <span class="cn">NULL</span>,
  regions <span class="op">=</span> <span class="cn">NULL</span>
<span class="op">)</span>

<span class="fu">junction_norm</span><span class="op">(</span><span class="va">junctions</span><span class="op">)</span>

<span class="fu">junction_process</span><span class="op">(</span>
  <span class="va">junctions</span>,
  <span class="va">ref</span>,
  ref_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene_id"</span>, <span class="st">"tx_name"</span>, <span class="st">"exon_name"</span><span class="op">)</span>,
  ref_cols_to_merge <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene_id"</span><span class="op">)</span>,
  count_thresh <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>raw <span class="op">=</span> <span class="fl">5</span><span class="op">)</span>,
  n_samp <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>raw <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>,
  width_range <span class="op">=</span> <span class="cn">NULL</span>,
  types <span class="op">=</span> <span class="cn">NULL</span>,
  regions <span class="op">=</span> <span class="cn">NULL</span>,
  score_func <span class="op">=</span> <span class="va">.zscore</span>,
  <span class="va">...</span>
<span class="op">)</span>

<span class="fu">junction_score</span><span class="op">(</span><span class="va">junctions</span>, score_func <span class="op">=</span> <span class="va">.zscore</span>, <span class="va">...</span><span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>junctions</dt>
<dd><p>junction data as a
<a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html" class="external-link">RangedSummarizedExperiment-class</a>
object.</p></dd>
<dt>ref</dt>
<dd><p>either path to gtf/gff3 or object of class
<a href="https://rdrr.io/pkg/GenomicFeatures/man/TxDb-class.html" class="external-link">TxDb-class</a> or
<a href="https://rdrr.io/pkg/ensembldb/man/EnsDb-class.html" class="external-link">EnsDb-class</a>.
<a href="https://rdrr.io/pkg/ensembldb/man/EnsDb-class.html" class="external-link">EnsDb-class</a> is required if you intend to annotate
junctions with gene symbols/names.</p></dd>
<dt>ref_cols</dt>
<dd><p>character vector listing the names of the columns in <code>ref</code>
for which to annotate junctions with. Must contain "gene_id", used for
categorising junctions.</p></dd>
<dt>ref_cols_to_merge</dt>
<dd><p>character vector listing which of the annotation
columns <code>ref_cols</code> should be merged into in columns to merge into a single
column per junction. Must contain "gene_id", used for categorising
junctions.</p></dd>
<dt>count_thresh</dt>
<dd><p>named vector with names matching the names of the
<a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a> in <code>junctions</code>.
Values denote the number of counts below which a junction will be filtered
out.</p></dd>
<dt>n_samp</dt>
<dd><p>named vector with names matching the names of the
<a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a> in <code>junctions</code>.
Values denotes number of samples that have to express the junction above
the <code>count_thresh</code> in order for that junction to not be filtered.</p></dd>
<dt>width_range</dt>
<dd><p>numeric vector of length 2. The first element denoting the
lower limit of junction width and the second the upper limit. Junctions
with widths outside this range will be filtered out.</p></dd>
<dt>types</dt>
<dd><p>any junctions matching these types, derived form
junction_annot will be filtered out.</p></dd>
<dt>regions</dt>
<dd><p>any junctions overlapping this set of regions (in a
<a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html" class="external-link">GRanges-class</a> format) will be filtered out.</p></dd>
<dt>score_func</dt>
<dd><p>function to score junctions by their abnormality. By
default, will use a z-score but can be switched to a user-defined function.
This function must take as input an <code>x</code> and <code>y</code> argument, containing case
and control counts respectively. This must return a numeric vector equal to
the length of <code>x</code> with elements corresponding to a abnormality of each
junction.</p></dd>
<dt>...</dt>
<dd><p>additional arguments passed to <code>score_func</code>.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p><a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html" class="external-link">RangedSummarizedExperiment-class</a>object containing filtered, annotated, normalised junction data with
abnormality scores.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p><code>junction_process</code> wraps all "junction_" prefixed functions in
<a href="dasper.html">dasper</a> except <a href="junction_load.html">junction_load</a>. This is designed to
simplify processing of the junction data for those familiar or uninterested
with the intermediates.</p>
<p><code>junction_annot</code> annotates junctions by 1. whether their start and/or end
position precisely overlaps with an annotated exon boundary and 2. whether
that junction matches an intron definition from existing annotation. Using
this information along with the strand, junctions are categorised into
"annotated", "novel_acceptor", "novel_donor", "novel_combo",
"novel_exon_skip", "ambig_gene" and "unannotated".</p>
<p><code>junction_filter</code> filters out "noisy" junctions based on counts, the width
of junctions, annotation category of the junction returned from
junction_annot and whether the junction overlaps with a set of
(blacklist) regions.</p>
<p><code>junction_norm</code> normalises the raw junction counts by 1. building junction
clusters by finding junctions that share an acceptor or donor position and
2. calculating a proportion-spliced-in (PSI) for each junction by dividing
the raw junction count by the total number of counts in it's associated
cluster.</p>
<p><code>junction_score</code> will use the counts contained within the "norm"
<a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a> to calculate a
deviation of each patient junction from the expected distribution of
control junction counts. The function used to calculate this abnormality
score can be user-inputted or left as the default z-score. Junctions will
also be labelled based on whether they are up-regulated (+1) or
down-regulated (-1) with respect to controls junction and this information
is stored in the <a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a>
"direction" for use in <a href="outlier_process.html">outlier_aggregate</a>.</p>
    </div>
    <div id="functions">
    <h2>Functions</h2>
    
<ul><li><p><code>junction_annot</code>: Annotate junctions using reference annotation</p></li>
<li><p><code>junction_filter</code>: Filter junctions by count, width, annotation or region</p></li>
<li><p><code>junction_norm</code>: Normalise junction counts by cluster</p></li>
<li><p><code>junction_score</code>: Score patient junctions by their abnormality</p></li>
</ul></div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p>ENCODE blacklist regions are recommended to be included as <code>regions</code>
for <code>junction_filter</code> and can be downloaded from
<a href="https://github.com/Boyle-Lab/Blacklist/blob/master/lists/hg38-blacklist.v2.bed.gz" class="external-link">https://github.com/Boyle-Lab/Blacklist/blob/master/lists/hg38-blacklist.v2.bed.gz</a>.
Further information can be found via the publication
<a href="https://www.nature.com/articles/s41598-019-45839-z" class="external-link">https://www.nature.com/articles/s41598-019-45839-z</a>.</p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"></span>
<span class="r-in"><span class="co">##### Set up txdb #####</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># use GenomicState to load txdb (GENCODE v31)</span></span>
<span class="r-in"><span class="va">ref</span> <span class="op">&lt;-</span> <span class="fu">GenomicState</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicState/man/GenomicStateHub.html" class="external-link">GenomicStateHub</a></span><span class="op">(</span></span>
<span class="r-in">    version <span class="op">=</span> <span class="st">"31"</span>,</span>
<span class="r-in">    genome <span class="op">=</span> <span class="st">"hg38"</span>,</span>
<span class="r-in">    filetype <span class="op">=</span> <span class="st">"TxDb"</span></span>
<span class="r-in"><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> snapshotDate(): 2021-10-20</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">##### junction_annot #####</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">junctions</span> <span class="op">&lt;-</span> <span class="fu">junction_annot</span><span class="op">(</span><span class="va">junctions_example</span>, <span class="va">ref</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:14:58 - Obtaining co-ordinates of annotated exons and junctions..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:11 - Getting junction annotation using overlapping exons..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:14 - Tidying junction annotation..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:14 - Deriving junction categories..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:19 - done!"</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">##### junction_filter #####</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">junctions</span> <span class="op">&lt;-</span> <span class="fu">junction_filter</span><span class="op">(</span></span>
<span class="r-in">    <span class="va">junctions</span>,</span>
<span class="r-in">    types <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ambig_gene"</span>, <span class="st">"unannotated"</span><span class="op">)</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:19 - Filtering junctions..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:19 - by count..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:19 - by type..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:19 - done!"</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">##### junction_norm #####</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">junctions</span> <span class="op">&lt;-</span> <span class="fu">junction_norm</span><span class="op">(</span><span class="va">junctions</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:19 - Clustering junctions..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:20 - Normalising junction counts..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:20 - done!"</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">##### junction_score #####</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">junctions</span> <span class="op">&lt;-</span> <span class="fu">junction_score</span><span class="op">(</span><span class="va">junctions</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:20 - Calculating the direction of change of junctions..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:20 - Generating junction abnormality score..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:20 - done!"</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">##### junction_process #####</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">junctions_processed</span> <span class="op">&lt;-</span> <span class="fu">junction_process</span><span class="op">(</span></span>
<span class="r-in">    <span class="va">junctions_example</span>,</span>
<span class="r-in">    <span class="va">ref</span>,</span>
<span class="r-in">    types <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ambig_gene"</span>, <span class="st">"unannotated"</span><span class="op">)</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "# Filtering junctions -----------------------------------------------------"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:20 - Filtering junctions..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:20 - by count..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:20 - done!"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "# Annotating junctions ----------------------------------------------------"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:20 - Obtaining co-ordinates of annotated exons and junctions..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:34 - Getting junction annotation using overlapping exons..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:35 - Tidying junction annotation..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:35 - Deriving junction categories..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:37 - done!"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "# Filtering junctions -----------------------------------------------------"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:37 - Filtering junctions..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:37 - by type..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:37 - done!"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "# Normalise junctions -----------------------------------------------------"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:37 - Clustering junctions..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:37 - Normalising junction counts..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:37 - done!"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "# Score junctions ---------------------------------------------------------"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:37 - Calculating the direction of change of junctions..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:37 - Generating junction abnormality score..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-03-03 20:15:38 - done!"</span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># the two objects are equivalent</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/all.equal.html" class="external-link">all.equal</a></span><span class="op">(</span><span class="va">junctions_processed</span>, <span class="va">junctions</span>, check.attributes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] TRUE</span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by David Zhang.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

