<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Processing outliers — outlier_aggregate • dasper</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Processing outliers — outlier_aggregate"><meta property="og:description" content='The set of functions prefixed with "outlier_" are used to detect
outliers. They are designed to be run after you have extracted your
junctions and coverage based features, in the order outlier_detect,
outlier_aggregate. Or, alternatively the wrapper function
outlier_process can be used to run the 2 functions stated above in one
go. For more details of the individual functions, see "Details".'><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">dasper</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.5.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/dasper.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/dzhang32/dasper/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Processing outliers</h1>
    <small class="dont-index">Source: <a href="https://github.com/dzhang32/dasper/blob/HEAD/R/outlier_aggregate.R" class="external-link"><code>R/outlier_aggregate.R</code></a>, <a href="https://github.com/dzhang32/dasper/blob/HEAD/R/outlier_detect.R" class="external-link"><code>R/outlier_detect.R</code></a>, <a href="https://github.com/dzhang32/dasper/blob/HEAD/R/outlier_process.R" class="external-link"><code>R/outlier_process.R</code></a></small>
    <div class="hidden name"><code>outlier_process.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The set of functions prefixed with "outlier_" are used to detect
outliers. They are designed to be run after you have extracted your
junctions and coverage based features, in the order <code>outlier_detect</code>,
<code>outlier_aggregate</code>. Or, alternatively the wrapper function
<code>outlier_process</code> can be used to run the 2 functions stated above in one
go. For more details of the individual functions, see "Details".</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">outlier_aggregate</span><span class="op">(</span>
  <span class="va">junctions</span>,
  samp_id_col <span class="op">=</span> <span class="st">"samp_id"</span>,
  bp_param <span class="op">=</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/SerialParam-class.html" class="external-link">SerialParam</a></span><span class="op">(</span><span class="op">)</span>
<span class="op">)</span>

<span class="fu">outlier_detect</span><span class="op">(</span>
  <span class="va">junctions</span>,
  feature_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"score"</span>, <span class="st">"coverage_score"</span><span class="op">)</span>,
  bp_param <span class="op">=</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/SerialParam-class.html" class="external-link">SerialParam</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="va">...</span>
<span class="op">)</span>

<span class="fu">outlier_process</span><span class="op">(</span>
  <span class="va">junctions</span>,
  feature_names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"score"</span>, <span class="st">"coverage_score"</span><span class="op">)</span>,
  samp_id_col <span class="op">=</span> <span class="st">"samp_id"</span>,
  bp_param <span class="op">=</span> <span class="fu">BiocParallel</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocParallel/man/SerialParam-class.html" class="external-link">SerialParam</a></span><span class="op">(</span><span class="op">)</span>,
  <span class="va">...</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>junctions</dt>
<dd><p>junction data as a
<a href="https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html" class="external-link">RangedSummarizedExperiment-class</a>
object.</p></dd>
<dt>samp_id_col</dt>
<dd><p>name of the column in the
<a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">SummarizedExperiment</a>
that details the sample ids.</p></dd>
<dt>bp_param</dt>
<dd><p>a
<a href="https://rdrr.io/pkg/BiocParallel/man/BiocParallelParam-class.html" class="external-link">BiocParallelParam-class</a> instance
denoting whether to parallelise the calculating of outlier scores across
samples.</p></dd>
<dt>feature_names</dt>
<dd><p>names of assays in <code>junctions</code> that are to be used as
input into the outlier detection model.</p></dd>
<dt>...</dt>
<dd><p>additional arguments passed to the outlier detection model
(isolation forest) for setting parameters.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p><code>DataFrame</code> with one row per cluster detailing each cluster's
associated junctions, outlier scores, ranks and genes.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p><code>outlier_process</code> wraps all "outlier_" prefixed functions in
<a href="dasper.html">dasper</a>. This is designed to simplify processing of the detecting outlier
junctions for those familiar or uninterested with the intermediates.</p>
<p><code>outlier_detect</code> will use the features in
<a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assays</a> named
<code>feature_names</code> as input into an unsupervised outlier detection algorithm
to score each junction based on how outlier-y it looks in relation to other
junctions in the patient. The default expected <code>score</code> and <code>coverage_score</code>
features can be calculated using the <a href="junction_process.html">junction_process</a> and
<a href="coverage_process.html">coverage_process</a> respectively.</p>
<p><code>outlier_aggregate</code> will aggregate the outlier scores into a cluster-level.
It will then rank each cluster based on this aggregated score and annotate
each cluster with it's associated gene and transcript.</p>
    </div>
    <div id="functions">
    <h2>Functions</h2>
    
<ul><li><p><code>outlier_aggregate</code>: Aggregate outlier scores from per junction to
cluster-level</p></li>
<li><p><code>outlier_detect</code>: Detecting outlier junctions</p></li>
</ul></div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p>for more details on the isolation forest model used:
https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.IsolationForest.html</p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="co"># \donttest{</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">##### Set up txdb #####</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># use GenomicState to load txdb (GENCODE v31)</span></span>
<span class="r-in"><span class="va">ref</span> <span class="op">&lt;-</span> <span class="fu">GenomicState</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicState/man/GenomicStateHub.html" class="external-link">GenomicStateHub</a></span><span class="op">(</span></span>
<span class="r-in">    version <span class="op">=</span> <span class="st">"31"</span>,</span>
<span class="r-in">    genome <span class="op">=</span> <span class="st">"hg38"</span>,</span>
<span class="r-in">    filetype <span class="op">=</span> <span class="st">"TxDb"</span></span>
<span class="r-in"><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> snapshotDate(): 2021-10-20</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> loading from cache</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">##### Set up BigWig #####</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># obtain path to example bw on recount2</span></span>
<span class="r-in"><span class="va">bw_path</span> <span class="op">&lt;-</span> <span class="fu">recount</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/recount/man/download_study.html" class="external-link">download_study</a></span><span class="op">(</span></span>
<span class="r-in">    project <span class="op">=</span> <span class="st">"SRP012682"</span>,</span>
<span class="r-in">    type <span class="op">=</span> <span class="st">"samples"</span>,</span>
<span class="r-in">    download <span class="op">=</span> <span class="cn">FALSE</span></span>
<span class="r-in"><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># cache the bw for speed in later</span></span>
<span class="r-in"><span class="co"># examples/testing during R CMD Check</span></span>
<span class="r-in"><span class="va">bw_path</span> <span class="op">&lt;-</span> <span class="fu">dasper</span><span class="fu">:::</span><span class="fu">.file_cache</span><span class="op">(</span><span class="va">bw_path</span><span class="op">)</span></span>
<span class="r-in"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">##### junction_process #####</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">junctions_processed</span> <span class="op">&lt;-</span> <span class="fu"><a href="junction_process.html">junction_process</a></span><span class="op">(</span></span>
<span class="r-in">    <span class="va">junctions_example</span>,</span>
<span class="r-in">    <span class="va">ref</span>,</span>
<span class="r-in">    types <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ambig_gene"</span>, <span class="st">"unannotated"</span><span class="op">)</span>,</span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "# Filtering junctions -----------------------------------------------------"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:26 - Filtering junctions..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:26 - by count..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:26 - done!"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "# Annotating junctions ----------------------------------------------------"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:26 - Obtaining co-ordinates of annotated exons and junctions..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:38 - Getting junction annotation using overlapping exons..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:39 - Tidying junction annotation..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:39 - Deriving junction categories..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:40 - done!"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "# Filtering junctions -----------------------------------------------------"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:40 - Filtering junctions..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:40 - by type..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:40 - done!"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "# Normalise junctions -----------------------------------------------------"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:40 - Clustering junctions..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:40 - Normalising junction counts..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:40 - done!"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "# Score junctions ---------------------------------------------------------"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:40 - Calculating the direction of change of junctions..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:40 - Generating junction abnormality score..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:40 - done!"</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">##### coverage_process #####</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">junctions_w_coverage</span> <span class="op">&lt;-</span> <span class="fu"><a href="coverage_process.html">coverage_process</a></span><span class="op">(</span></span>
<span class="r-in">    <span class="va">junctions_processed</span>,</span>
<span class="r-in">    <span class="va">ref</span>,</span>
<span class="r-in">    coverage_paths_case <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="va">bw_path</span>, <span class="fl">2</span><span class="op">)</span>,</span>
<span class="r-in">    coverage_paths_control <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="va">bw_path</span>, <span class="fl">3</span><span class="op">)</span></span>
<span class="r-in"><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "# Loading and normalising coverage ---------------------------------------------"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:40 - Obtaining exonic and intronic regions to load coverage from..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:41 - Obtaining regions to use to normalise coverage..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:46 - Loading coverage..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:58 - Normalising coverage..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:58 - done!"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "# Scoring coverage ---------------------------------------------"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:58 - Generating coverage abnormality score..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:59 - Obtaining regions with greatest coverage dysruption..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:59 - done!"</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">##### outlier_detect #####</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">junctions_w_outliers</span> <span class="op">&lt;-</span> <span class="fu">outlier_detect</span><span class="op">(</span><span class="va">junctions_w_coverage</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:03:59 - generating outlier scores for sample 1/2"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:04:01 - fitting outlier detection model with parameters: behaviour=deprecated, bootstrap=FALSE, contamination=auto, max_features=1, max_samples=auto, n_estimators=100, n_jobs=NULL, random_state=NULL, verbose=0, warm_start=FALSE"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:04:01 - fitting outlier detection model with parameters: behaviour=deprecated, bootstrap=FALSE, contamination=auto, max_features=1, max_samples=auto, n_estimators=100, n_jobs=NULL, random_state=NULL, verbose=0, warm_start=FALSE"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:04:01 - generating outlier scores for sample 2/2"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:04:02 - fitting outlier detection model with parameters: behaviour=deprecated, bootstrap=FALSE, contamination=auto, max_features=1, max_samples=auto, n_estimators=100, n_jobs=NULL, random_state=NULL, verbose=0, warm_start=FALSE"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:04:02 - fitting outlier detection model with parameters: behaviour=deprecated, bootstrap=FALSE, contamination=auto, max_features=1, max_samples=auto, n_estimators=100, n_jobs=NULL, random_state=NULL, verbose=0, warm_start=FALSE"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:04:02 - done!"</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">##### outlier_aggregate #####</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">outlier_scores</span> <span class="op">&lt;-</span> <span class="fu">outlier_aggregate</span><span class="op">(</span><span class="va">junctions_w_outliers</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:04:02 - Aggregating outlier scores to cluster level... "</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:04:03 - Annotating clusters with gene details..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:04:03 - done!"</span>
<span class="r-in"></span>
<span class="r-in"><span class="co">##### outlier_process #####</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># this wrapper will obtain outlier scores identical to those</span></span>
<span class="r-in"><span class="co"># obtained through running the individual wrapped functions shown below</span></span>
<span class="r-in"><span class="va">outlier_processed</span> <span class="op">&lt;-</span> <span class="fu">outlier_process</span><span class="op">(</span><span class="va">junctions_w_coverage</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "# Detecting outliers using an isolation forest ------------------------------"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:04:03 - generating outlier scores for sample 1/2"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:04:03 - fitting outlier detection model with parameters: behaviour=deprecated, bootstrap=FALSE, contamination=auto, max_features=1, max_samples=auto, n_estimators=100, n_jobs=NULL, random_state=NULL, verbose=0, warm_start=FALSE"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:04:04 - fitting outlier detection model with parameters: behaviour=deprecated, bootstrap=FALSE, contamination=auto, max_features=1, max_samples=auto, n_estimators=100, n_jobs=NULL, random_state=NULL, verbose=0, warm_start=FALSE"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:04:04 - generating outlier scores for sample 2/2"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:04:04 - fitting outlier detection model with parameters: behaviour=deprecated, bootstrap=FALSE, contamination=auto, max_features=1, max_samples=auto, n_estimators=100, n_jobs=NULL, random_state=NULL, verbose=0, warm_start=FALSE"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:04:04 - fitting outlier detection model with parameters: behaviour=deprecated, bootstrap=FALSE, contamination=auto, max_features=1, max_samples=auto, n_estimators=100, n_jobs=NULL, random_state=NULL, verbose=0, warm_start=FALSE"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:04:04 - done!"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "# Aggregating outlier scores to cluster-level -----------------------------"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:04:04 - Aggregating outlier scores to cluster level... "</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:04:05 - Annotating clusters with gene details..."</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "2022-01-10 23:04:05 - done!"</span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># the two objects are equivalent</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/base/all.equal.html" class="external-link">all.equal</a></span><span class="op">(</span><span class="va">outlier_processed</span>, <span class="va">outlier_scores</span>, check.attributes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] TRUE</span>
<span class="r-in"><span class="co"># }</span></span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by David Zhang.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.9000.</p>
</div>

      </footer></div>

  


  

  </body></html>

