<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Visualise RNA-seq data in a the form of a sashimi plot — plot_sashimi • dasper</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Visualise RNA-seq data in a the form of a sashimi plot — plot_sashimi" />
<meta property="og:description" content="plot_sashimi plots the splicing events and coverage across specific
genes/transcripts/regions of interest. Unlike traditional sashimi plots,
coverage and junction tracks are separated, which enables user's to choose
whether they would like to plot only the junctions." />






<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  

  

  </head>

  <body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">dasper</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.3.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/dasper.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/dzhang32/dasper/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Visualise RNA-seq data in a the form of a sashimi plot</h1>
    <small class="dont-index">Source: <a href='https://github.com/dzhang32/dasper/blob/master/R/plot_sashimi.R'><code>R/plot_sashimi.R</code></a></small>
    <div class="hidden name"><code>plot_sashimi.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>plot_sashimi</code> plots the splicing events and coverage across specific
genes/transcripts/regions of interest. Unlike traditional sashimi plots,
coverage and junction tracks are separated, which enables user's to choose
whether they would like to plot only the junctions.</p>
    </div>

    <div class="ref-usage sourceCode"><pre class='sourceCode r'><code><span class='fu'>plot_sashimi</span><span class='op'>(</span>
  <span class='va'>junctions</span>,
  <span class='va'>ref</span>,
  <span class='va'>gene_tx_id</span>,
  <span class='va'>gene_tx_col</span>,
  case_id <span class='op'>=</span> <span class='cn'>NULL</span>,
  sum_func <span class='op'>=</span> <span class='va'>mean</span>,
  region <span class='op'>=</span> <span class='cn'>NULL</span>,
  assay_name <span class='op'>=</span> <span class='st'>"norm"</span>,
  annot_colour <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fu'>ggpubr</span><span class='fu'>::</span><span class='fu'><a href='https://rpkgs.datanovia.com/ggpubr/reference/get_palette.html'>get_palette</a></span><span class='op'>(</span><span class='st'>"jco"</span>, <span class='fl'>1</span><span class='op'>)</span>, <span class='fu'>ggpubr</span><span class='fu'>::</span><span class='fu'><a href='https://rpkgs.datanovia.com/ggpubr/reference/get_palette.html'>get_palette</a></span><span class='op'>(</span><span class='st'>"npg"</span>, <span class='fl'>7</span><span class='op'>)</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>3</span>,
    <span class='fl'>2</span>, <span class='fl'>5</span>, <span class='fl'>6</span><span class='op'>)</span><span class='op'>]</span>, <span class='fu'>ggpubr</span><span class='fu'>::</span><span class='fu'><a href='https://rpkgs.datanovia.com/ggpubr/reference/get_palette.html'>get_palette</a></span><span class='op'>(</span><span class='st'>"jco"</span>, <span class='fl'>6</span><span class='op'>)</span><span class='op'>[</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>3</span><span class='op'>)</span><span class='op'>]</span><span class='op'>)</span>,
  digits <span class='op'>=</span> <span class='fl'>2</span>,
  count_label <span class='op'>=</span> <span class='cn'>TRUE</span>,
  coverage_paths_case <span class='op'>=</span> <span class='cn'>NULL</span>,
  coverage_paths_control <span class='op'>=</span> <span class='cn'>NULL</span>,
  coverage_chr_control <span class='op'>=</span> <span class='cn'>NULL</span>,
  load_func <span class='op'>=</span> <span class='va'>.coverage_load</span>,
  binwidth <span class='op'>=</span> <span class='fl'>100</span>
<span class='op'>)</span></code></pre></div>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>junctions</th>
      <td><p>junction data as a
<a href='https://rdrr.io/pkg/SummarizedExperiment/man/RangedSummarizedExperiment-class.html'>RangedSummarizedExperiment-class</a>
object.</p></td>
    </tr>
    <tr>
      <th>ref</th>
      <td><p>either path to gtf/gff3 or object of class
<a href='https://rdrr.io/pkg/GenomicFeatures/man/TxDb-class.html'>TxDb-class</a> or
<a href='https://rdrr.io/pkg/ensembldb/man/EnsDb-class.html'>EnsDb-class</a>.
<a href='https://rdrr.io/pkg/ensembldb/man/EnsDb-class.html'>EnsDb-class</a> is required if you intend to annotate
junctions with gene symbols/names.</p></td>
    </tr>
    <tr>
      <th>gene_tx_id</th>
      <td><p>character scalar with the id of the gene. This must be a an
identifier for a gene or transcript, which has a matching entry in <code>ref</code>.</p></td>
    </tr>
    <tr>
      <th>gene_tx_col</th>
      <td><p>character scalar with the name of the column to search for
the <code>gene_tx_id</code> in <code>ref</code>.</p></td>
    </tr>
    <tr>
      <th>case_id</th>
      <td><p>list containing 1 element. The contents of this element must
be a character vector specifying sample ids that are to be plotted. The
name of this element must correspond to the column containing sample ids in
the junction <code>SummarizedExperiment::mcols()</code>. By default, all cases will be
plotted.</p></td>
    </tr>
    <tr>
      <th>sum_func</th>
      <td><p>function that will be used to aggregate the junction counts
and coverage for controls. By default, <code>mean</code> will be used.</p></td>
    </tr>
    <tr>
      <th>region</th>
      <td><p>a <a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GenomicRanges</a> of length 1
that is used to filter the exons/junctions plotted. Only those that overlap
this region are plotted.</p></td>
    </tr>
    <tr>
      <th>assay_name</th>
      <td><p>a character scalar with the name of the
<code><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>SummarizedExperiment::assay()</a></code> from which to obtain junction counts.</p></td>
    </tr>
    <tr>
      <th>annot_colour</th>
      <td><p>character vector length 7, representing the colours of
junction types.</p></td>
    </tr>
    <tr>
      <th>digits</th>
      <td><p>used in <code><a href='https://rdrr.io/r/base/Round.html'>round()</a></code>, specifying the number of digits to round the
junction counts to for visualisation purposes.</p></td>
    </tr>
    <tr>
      <th>count_label</th>
      <td><p>logical value specifying whether to add label the count of
each junction.</p></td>
    </tr>
    <tr>
      <th>coverage_paths_case</th>
      <td><p>paths to the BigWig files containing the coverage
of your case samples.</p></td>
    </tr>
    <tr>
      <th>coverage_paths_control</th>
      <td><p>paths to the BigWig files for control samples.</p></td>
    </tr>
    <tr>
      <th>coverage_chr_control</th>
      <td><p>either "chr" or "no_chr", indicating the
chromosome format of control coverage data. Only required if the
chromosome format of the control BigWig files is different to that of your
cases.</p></td>
    </tr>
    <tr>
      <th>load_func</th>
      <td><p>function used to load coverage.</p></td>
    </tr>
    <tr>
      <th>binwidth</th>
      <td><p>the number of bases to aggregate coverage across using
<code>sum_func</code> when plotting. .</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p><code>ggplot</code> displaying the splicing (and coverage) surrounding the
transcript/region of interest.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <div class="ref-examples sourceCode"><pre class='sourceCode r'><code><span class='r-in'></span>
<span class='r-in'><span class='co'># use GenomicState to load txdb (GENCODE v31)</span></span>
<span class='r-in'><span class='va'>ref</span> <span class='op'>&lt;-</span> <span class='fu'>GenomicState</span><span class='fu'>::</span><span class='fu'><a href='https://rdrr.io/pkg/GenomicState/man/GenomicStateHub.html'>GenomicStateHub</a></span><span class='op'>(</span></span>
<span class='r-in'>    version <span class='op'>=</span> <span class='st'>"31"</span>,</span>
<span class='r-in'>    genome <span class='op'>=</span> <span class='st'>"hg38"</span>,</span>
<span class='r-in'>    filetype <span class='op'>=</span> <span class='st'>"TxDb"</span></span>
<span class='r-in'><span class='op'>)</span><span class='op'>[[</span><span class='fl'>1</span><span class='op'>]</span><span class='op'>]</span></span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> snapshotDate(): 2021-09-10</span>
<span class='r-msg co'><span class='r-pr'>#&gt;</span> loading from cache</span>
<span class='r-in'></span>
<span class='r-in'><span class='va'>junctions_processed</span> <span class='op'>&lt;-</span> <span class='fu'><a href='junction_process.html'>junction_process</a></span><span class='op'>(</span></span>
<span class='r-in'>    <span class='va'>junctions_example</span>,</span>
<span class='r-in'>    <span class='va'>ref</span>,</span>
<span class='r-in'>    types <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"ambig_gene"</span>, <span class='st'>"unannotated"</span><span class='op'>)</span></span>
<span class='r-in'><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "# Filtering junctions -----------------------------------------------------"</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "2021-09-21 16:51:54 - Filtering junctions..."</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "2021-09-21 16:51:54 - by count..."</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "2021-09-21 16:51:55 - done!"</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "# Annotating junctions ----------------------------------------------------"</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "2021-09-21 16:51:55 - Obtaining co-ordinates of annotated exons and junctions..."</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "2021-09-21 16:52:10 - Getting junction annotation using overlapping exons..."</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "2021-09-21 16:52:11 - Tidying junction annotation..."</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "2021-09-21 16:52:12 - Deriving junction categories..."</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "2021-09-21 16:52:12 - done!"</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "# Filtering junctions -----------------------------------------------------"</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "2021-09-21 16:52:12 - Filtering junctions..."</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "2021-09-21 16:52:12 - by type..."</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "2021-09-21 16:52:12 - done!"</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "# Normalise junctions -----------------------------------------------------"</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "2021-09-21 16:52:12 - Clustering junctions..."</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "2021-09-21 16:52:13 - Normalising junction counts..."</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "2021-09-21 16:52:13 - done!"</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "# Score junctions ---------------------------------------------------------"</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "2021-09-21 16:52:13 - Calculating the direction of change of junctions..."</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "2021-09-21 16:52:13 - Generating junction abnormality score..."</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "2021-09-21 16:52:13 - done!"</span>
<span class='r-in'></span>
<span class='r-in'><span class='va'>sashimi_plot</span> <span class='op'>&lt;-</span> <span class='fu'>plot_sashimi</span><span class='op'>(</span></span>
<span class='r-in'>    junctions <span class='op'>=</span> <span class='fu'><a href='junction_process.html'>junction_filter</a></span><span class='op'>(</span><span class='va'>junctions_processed</span><span class='op'>)</span>,</span>
<span class='r-in'>    ref <span class='op'>=</span> <span class='va'>ref</span>,</span>
<span class='r-in'>    gene_tx_id <span class='op'>=</span> <span class='st'>"ENSG00000142156.14"</span>,</span>
<span class='r-in'>    gene_tx_col <span class='op'>=</span> <span class='st'>"gene_id"</span>,</span>
<span class='r-in'>    sum_func <span class='op'>=</span> <span class='cn'>NULL</span></span>
<span class='r-in'><span class='op'>)</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "2021-09-21 16:52:13 - Filtering junctions..."</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "2021-09-21 16:52:13 - by count..."</span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] "2021-09-21 16:52:14 - done!"</span>
</code></pre></div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p><p>Developed by David Zhang.</p></p>
</div>

<div class="pkgdown">
  <p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 1.6.1.9001.</p></p>
</div>

      </footer>
   </div>

  


  

  </body>
</html>


