<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to dasper • dasper</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to dasper">
<meta property="og:description" content="dasper">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">dasper</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/dasper.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/dzhang32/dasper/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="dasper_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to dasper</h1>
                        <h4 class="author">David Zhang</h4>
            <address class="author_afil">
      UCL<br><a class="author_email" href="mailto:#"></a><a href="mailto:david.zhang.12@ucl.ac.uk" class="email">david.zhang.12@ucl.ac.uk</a>
      </address>
                  
            <h4 class="date">4 August 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/dzhang32/dasper/blob/master/vignettes/dasper.Rmd"><code>vignettes/dasper.Rmd</code></a></small>
      <div class="hidden name"><code>dasper.Rmd</code></div>

    </div>

    
    
<div id="basics" class="section level1">
<h1 class="hasAnchor">
<a href="#basics" class="anchor"></a>Basics</h1>
<div id="install-dasper" class="section level2">
<h2 class="hasAnchor">
<a href="#install-dasper" class="anchor"></a>Install <code>dasper</code>
</h2>
<p><code>R</code> is an open-source statistical environment which can be easily modified to enhance its functionality via packages. <em><a href="https://bioconductor.org/packages/3.12/dasper">dasper</a></em> is a <code>R</code> package available via the <a href="http://bioconductor.org">Bioconductor</a> repository for packages. <code>R</code> can be installed on any operating system from <a href="https://cran.r-project.org/">CRAN</a> after which you can install <em><a href="https://bioconductor.org/packages/3.12/dasper">dasper</a></em> by using the following commands in your <code>R</code> session:</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="co">if</span> (<span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"BiocManager"</span>, quietly = <span class="fl">TRUE</span>)) {
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"BiocManager"</span>)
}

<span class="kw">BiocManager</span>::<span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"dasper"</span>)

<span class="co">## Check that you have a valid Bioconductor installation</span>
<span class="kw">BiocManager</span>::<span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/valid.html">valid</a></span>()
</pre></div>
</div>
<div id="required-knowledge" class="section level2">
<h2 class="hasAnchor">
<a href="#required-knowledge" class="anchor"></a>Required knowledge</h2>
<p>The expected input of <em><a href="https://bioconductor.org/packages/3.12/dasper">dasper</a></em> are junctions reads (outputted from an aligner such as <a href="https://github.com/alexdobin/STAR">STAR</a>) and coverage in the form of BigWig files (which can be generated from BAM files using <a href="http://rseqc.sourceforge.net/#bam2wig-py">RSeQC</a>.<em><a href="https://bioconductor.org/packages/3.12/dasper">dasper</a></em> is based on many other packages and in particular in those that have implemented the infrastructure needed for dealing with RNA-seq data. The packages <em><a href="https://bioconductor.org/packages/3.12/SummarizedExperiment">SummarizedExperiment</a></em> and <em><a href="https://bioconductor.org/packages/3.12/GenomicRanges">GenomicRanges</a></em> are used throughout, therefore familiarity with these packages will greatly help in interpreting the output of <em><a href="https://bioconductor.org/packages/3.12/dasper">dasper</a></em>.</p>
<p>If you are asking yourself the question “Where do I start using Bioconductor?” you might be interested in <a href="http://lcolladotor.github.io/2014/10/16/startbioc/#.VkOKbq6rRuU">this blog post</a>.</p>
</div>
<div id="asking-for-help" class="section level2">
<h2 class="hasAnchor">
<a href="#asking-for-help" class="anchor"></a>Asking for help</h2>
<p>As package developers, we try to explain clearly how to use our packages and in which order to use the functions. But <code>R</code> and <code>Bioconductor</code> have a steep learning curve so it is critical to learn where to ask for help. The blog post quoted above mentions some but we would like to highlight the <a href="https://support.bioconductor.org/">Bioconductor support site</a> as the main resource for getting help: remember to use the <code>dasper</code> tag and check <a href="https://support.bioconductor.org/t/dasper/">the older posts</a>. Other alternatives are available such as creating GitHub issues and tweeting. However, please note that if you want to receive help you should adhere to the <a href="http://www.bioconductor.org/help/support/posting-guide/">posting guidelines</a>. It is particularly critical that you provide a small reproducible example and your session information so package developers can track down the source of the error.</p>
</div>
<div id="citing-dasper" class="section level2">
<h2 class="hasAnchor">
<a href="#citing-dasper" class="anchor"></a>Citing <code>dasper</code>
</h2>
<p>We hope that <em><a href="https://bioconductor.org/packages/3.12/dasper">dasper</a></em> will be useful for your research. Please use the following information to cite the package and the overall approach. Thank you!</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="co">## Citation info</span>
<span class="fu"><a href="https://rdrr.io/r/utils/citation.html">citation</a></span>(<span class="st">"dasper"</span>)
<span class="co">#&gt; </span>
<span class="co">#&gt; dzhang32 (2020). _Detecting abberant splicing events from</span>
<span class="co">#&gt; RNA-sequencing data_. doi: 10.18129/B9.bioc.dasper (URL:</span>
<span class="co">#&gt; https://doi.org/10.18129/B9.bioc.dasper),</span>
<span class="co">#&gt; https://github.com/dzhang32/dasper - R package version 0.99.0, &lt;URL:</span>
<span class="co">#&gt; http://www.bioconductor.org/packages/dasper&gt;.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; dzhang32 (2020). "Detecting abberant splicing events from</span>
<span class="co">#&gt; RNA-sequencing data." _bioRxiv_. doi: 10.1101/TODO (URL:</span>
<span class="co">#&gt; https://doi.org/10.1101/TODO), &lt;URL:</span>
<span class="co">#&gt; https://www.biorxiv.org/content/10.1101/TODO&gt;.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; To see these entries in BibTeX format, use 'print(&lt;citation&gt;,</span>
<span class="co">#&gt; bibtex=TRUE)', 'toBibtex(.)', or set</span>
<span class="co">#&gt; 'options(citation.bibtex.max=999)'.</span>
</pre></div>
</div>
</div>
<div id="quick-start-to-using-to-dasper" class="section level1">
<h1 class="hasAnchor">
<a href="#quick-start-to-using-to-dasper" class="anchor"></a>Quick start to using to <code>dasper</code>
</h1>
<div id="workflow" class="section level2">
<h2 class="hasAnchor">
<a href="#workflow" class="anchor"></a>Workflow</h2>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="kw">knitr</span>::<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/include_graphics.html">include_graphics</a></span>(<span class="st">"dasper_workflow.png"</span>)
</pre></div>
<p><img src="dasper_workflow.png"><!-- --></p>
<p>The above workflow diagram gives a top-level overview of the core functions within <code>dasper</code> and describes the order in which they are intended to be run. This are broadly split into 4 categories:</p>
<ol style="list-style-type: decimal">
<li>
<strong>Process junctions</strong> - these functions are prefixed with a “junction_”. They will load in your junctions to an <em><a href="https://bioconductor.org/packages/3.12/SummarizedExperiment">SummarizedExperiment</a></em>, annotate your junctions using reference annotation, filter out junctions for those that likely originate from techinical noise and normalise your junction counts to allow for comparison between samples.</li>
<li>
<strong>Process coverage</strong> - these functions are prefixed with a “cov_”. They annotate your junctions with coverage from nearby regions then normalise this coverage to allow for comparison between samples.</li>
<li>
<strong>Outlier detection</strong> - to be added.</li>
<li>
<strong>Sashimi plots</strong> - to be added.</li>
</ol>
</div>
<div id="example" class="section level2">
<h2 class="hasAnchor">
<a href="#example" class="anchor"></a>Example</h2>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://github.com/dzhang32/dasper">"dasper"</a></span>)
</pre></div>
<p>Edit this as you see fit =) (to be completed…)</p>
<p>Here is an example of you can cite your package inside the vignette:</p>
<ul>
<li>
<em><a href="https://bioconductor.org/packages/3.12/dasper">dasper</a></em> <a id="cite-dzhang32_2020"></a>(<a href="http://www.bioconductor.org/packages/dasper">dzhang32, 2020</a>)</li>
</ul>
</div>
</div>
<div id="reproducibility" class="section level1">
<h1 class="hasAnchor">
<a href="#reproducibility" class="anchor"></a>Reproducibility</h1>
<p>The <em><a href="https://bioconductor.org/packages/3.12/dasper">dasper</a></em> package (<a href="http://www.bioconductor.org/packages/dasper">dzhang32, 2020</a>) was made possible thanks to:</p>
<ul>
<li>R <a id="cite-R_Core_Team_2020"></a>(<a href="https://www.R-project.org/">R Core Team, 2020</a>)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.12/BiocStyle">BiocStyle</a></em> <a id="cite-Oles_2020"></a>(<a href="https://github.com/Bioconductor/BiocStyle">Oleś, Morgan, and Huber, 2020</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitcitations">knitcitations</a></em> <a id="cite-Boettiger_2019"></a>(<a href="https://github.com/cboettig/knitcitations">Boettiger, 2019</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr">knitr</a></em> <a id="cite-Xie_2020"></a>(<a href="https://yihui.org/knitr/">Xie, 2020</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown">rmarkdown</a></em> <a id="cite-Allaire_2020"></a>(<a href="https://github.com/rstudio/rmarkdown">Allaire, Xie, McPherson, Luraschi, et al., 2020</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo">sessioninfo</a></em> <a id="cite-Csardi_2018"></a>(<a href="https://github.com/r-lib/sessioninfo#readme">Csárdi, core, Wickham, Chang, et al., 2018</a>)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat">testthat</a></em> <a id="cite-Wickham_2011"></a>(<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf">Wickham, 2011</a>)</li>
</ul>
<p>This package was developed using <em><a href="https://github.com/lcolladotor/biocthis">biocthis</a></em>.</p>
<p>Code for creating the vignette</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="co">## Create the vignette</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://github.com/rstudio/rmarkdown">"rmarkdown"</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/rmarkdown/man/render.html">render</a></span>(<span class="st">"dasper.Rmd"</span>, <span class="st">"BiocStyle::html_document"</span>))

<span class="co">## Extract the R code</span>
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st"><a href="https://yihui.org/knitr">"knitr"</a></span>)
<span class="fu"><a href="https://rdrr.io/pkg/knitr/man/knit.html">knit</a></span>(<span class="st">"dasper.Rmd"</span>, tangle = <span class="fl">TRUE</span>)
</pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="co">## Clean up</span>
<span class="fu"><a href="https://rdrr.io/r/base/files.html">file.remove</a></span>(<span class="st">"dasper.bib"</span>)
<span class="co">#&gt; [1] TRUE</span>
</pre></div>
<p>Date the vignette was generated.</p>
<pre><code>#&gt; [1] "2020-08-04 11:59:21 UTC"</code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code>#&gt; Time difference of 5.972 secs</code></pre>
<p><code>R</code> session information.</p>
<pre><code>#&gt; ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  setting  value                       
#&gt;  version  R version 4.0.2 (2020-06-22)
#&gt;  os       Ubuntu 20.04.1 LTS          
#&gt;  system   x86_64, linux-gnu           
#&gt;  ui       X11                         
#&gt;  language (EN)                        
#&gt;  collate  en_US.UTF-8                 
#&gt;  ctype    en_US.UTF-8                 
#&gt;  tz       UTC                         
#&gt;  date     2020-08-04                  
#&gt; 
#&gt; ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  package              * version    date       lib source                        
#&gt;  assertthat             0.2.1      2019-03-21 [2] RSPM (R 4.0.0)                
#&gt;  backports              1.1.8      2020-06-17 [2] RSPM (R 4.0.1)                
#&gt;  bibtex                 0.4.2.2    2020-01-02 [1] RSPM (R 4.0.0)                
#&gt;  Biobase                2.49.0     2020-04-27 [1] Bioconductor                  
#&gt;  BiocGenerics           0.35.4     2020-06-04 [1] Bioconductor                  
#&gt;  BiocManager            1.30.10    2019-11-16 [2] CRAN (R 4.0.2)                
#&gt;  BiocStyle            * 2.17.0     2020-04-27 [1] Bioconductor                  
#&gt;  bitops                 1.0-6      2013-08-17 [1] RSPM (R 4.0.0)                
#&gt;  bookdown               0.20       2020-06-23 [1] RSPM (R 4.0.2)                
#&gt;  cli                    2.0.2      2020-02-28 [2] RSPM (R 4.0.0)                
#&gt;  crayon                 1.3.4      2017-09-16 [2] RSPM (R 4.0.0)                
#&gt;  dasper               * 0.99.0     2020-08-04 [1] Bioconductor                  
#&gt;  DelayedArray           0.15.7     2020-07-14 [1] Bioconductor                  
#&gt;  desc                   1.2.0      2018-05-01 [2] RSPM (R 4.0.0)                
#&gt;  digest                 0.6.25     2020-02-23 [2] RSPM (R 4.0.0)                
#&gt;  dplyr                  1.0.0      2020-05-29 [1] RSPM (R 4.0.0)                
#&gt;  ellipsis               0.3.1      2020-05-15 [2] RSPM (R 4.0.0)                
#&gt;  evaluate               0.14       2019-05-28 [2] RSPM (R 4.0.0)                
#&gt;  fansi                  0.4.1      2020-01-08 [2] RSPM (R 4.0.0)                
#&gt;  fs                     1.4.2      2020-06-30 [2] RSPM (R 4.0.2)                
#&gt;  generics               0.0.2      2018-11-29 [1] RSPM (R 4.0.0)                
#&gt;  GenomeInfoDb           1.25.8     2020-07-03 [1] Bioconductor                  
#&gt;  GenomeInfoDbData       1.2.3      2020-07-06 [1] Bioconductor                  
#&gt;  GenomicRanges          1.41.5     2020-06-09 [1] Bioconductor                  
#&gt;  glue                   1.4.1      2020-05-13 [2] RSPM (R 4.0.0)                
#&gt;  htmltools              0.5.0      2020-06-16 [2] RSPM (R 4.0.1)                
#&gt;  httr                   1.4.2      2020-07-20 [2] RSPM (R 4.0.2)                
#&gt;  IRanges                2.23.10    2020-06-13 [1] Bioconductor                  
#&gt;  jsonlite               1.7.0      2020-06-25 [2] RSPM (R 4.0.2)                
#&gt;  knitcitations        * 1.0.10     2019-09-15 [1] RSPM (R 4.0.0)                
#&gt;  knitr                  1.29       2020-06-23 [2] RSPM (R 4.0.2)                
#&gt;  lattice                0.20-41    2020-04-02 [3] CRAN (R 4.0.2)                
#&gt;  lifecycle              0.2.0      2020-03-06 [2] RSPM (R 4.0.0)                
#&gt;  lubridate              1.7.9      2020-06-08 [1] RSPM (R 4.0.2)                
#&gt;  magrittr               1.5        2014-11-22 [2] RSPM (R 4.0.0)                
#&gt;  Matrix                 1.2-18     2019-11-27 [3] CRAN (R 4.0.2)                
#&gt;  matrixStats            0.56.0     2020-03-13 [1] RSPM (R 4.0.0)                
#&gt;  memoise                1.1.0      2017-04-21 [2] RSPM (R 4.0.0)                
#&gt;  pillar                 1.4.6      2020-07-10 [2] RSPM (R 4.0.2)                
#&gt;  pkgconfig              2.0.3      2019-09-22 [2] RSPM (R 4.0.0)                
#&gt;  pkgdown                1.5.1.9000 2020-07-25 [1] Github (r-lib/pkgdown@eae56f0)
#&gt;  plyr                   1.8.6      2020-03-03 [1] RSPM (R 4.0.0)                
#&gt;  purrr                  0.3.4      2020-04-17 [2] RSPM (R 4.0.0)                
#&gt;  R6                     2.4.1      2019-11-12 [2] RSPM (R 4.0.0)                
#&gt;  Rcpp                   1.0.5      2020-07-06 [2] RSPM (R 4.0.2)                
#&gt;  RCurl                  1.98-1.2   2020-04-18 [1] RSPM (R 4.0.0)                
#&gt;  RefManageR             1.2.12     2019-04-03 [1] RSPM (R 4.0.0)                
#&gt;  rlang                  0.4.7      2020-07-09 [2] RSPM (R 4.0.2)                
#&gt;  rmarkdown              2.3        2020-06-18 [1] RSPM (R 4.0.1)                
#&gt;  rprojroot              1.3-2      2018-01-03 [2] RSPM (R 4.0.0)                
#&gt;  S4Vectors              0.27.12    2020-06-09 [1] Bioconductor                  
#&gt;  sessioninfo          * 1.1.1      2018-11-05 [2] RSPM (R 4.0.0)                
#&gt;  stringi                1.4.6      2020-02-17 [2] RSPM (R 4.0.0)                
#&gt;  stringr                1.4.0      2019-02-10 [2] RSPM (R 4.0.0)                
#&gt;  SummarizedExperiment   1.19.6     2020-07-09 [1] Bioconductor                  
#&gt;  tibble                 3.0.3      2020-07-10 [2] RSPM (R 4.0.2)                
#&gt;  tidyselect             1.1.0      2020-05-11 [1] RSPM (R 4.0.0)                
#&gt;  vctrs                  0.3.2      2020-07-15 [2] RSPM (R 4.0.2)                
#&gt;  withr                  2.2.0      2020-04-20 [2] RSPM (R 4.0.0)                
#&gt;  xfun                   0.16       2020-07-24 [2] RSPM (R 4.0.2)                
#&gt;  xml2                   1.3.2      2020-04-23 [2] RSPM (R 4.0.0)                
#&gt;  XVector                0.29.3     2020-06-25 [1] Bioconductor                  
#&gt;  yaml                   2.2.1      2020-02-01 [2] RSPM (R 4.0.0)                
#&gt;  zlibbioc               1.35.0     2020-04-27 [1] Bioconductor                  
#&gt; 
#&gt; [1] /usr/local/lib/R/host-site-library
#&gt; [2] /usr/local/lib/R/site-library
#&gt; [3] /usr/local/lib/R/library</code></pre>
</div>
<div id="bibliography" class="section level1">
<h1 class="hasAnchor">
<a href="#bibliography" class="anchor"></a>Bibliography</h1>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.12/BiocStyle">BiocStyle</a></em> (<a href="https://github.com/Bioconductor/BiocStyle">Oleś, Morgan, and Huber, 2020</a>) with <em><a href="https://CRAN.R-project.org/package=knitr">knitr</a></em> (<a href="https://yihui.org/knitr/">Xie, 2020</a>) and <em><a href="https://CRAN.R-project.org/package=rmarkdown">rmarkdown</a></em> (<a href="https://github.com/rstudio/rmarkdown">Allaire, Xie, McPherson, Luraschi, et al., 2020</a>) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=knitcitations">knitcitations</a></em> (<a href="https://github.com/cboettig/knitcitations">Boettiger, 2019</a>).</p>
<p>[1] J. Allaire, Y. Xie, J. McPherson, J. Luraschi, et al. <em>rmarkdown: Dynamic Documents for R</em>. R package version 2.3. 2020. &lt;URL: <a href="https://github.com/rstudio/rmarkdown" class="uri">https://github.com/rstudio/rmarkdown</a>&gt;.</p>
<p>[2] C. Boettiger. <em>knitcitations: Citations for ‘Knitr’ Markdown Files</em>. R package version 1.0.10. 2019. &lt;URL: <a href="https://github.com/cboettig/knitcitations" class="uri">https://github.com/cboettig/knitcitations</a>&gt;.</p>
<p>[3] G. Csárdi, R. core, H. Wickham, W. Chang, et al. <em>sessioninfo: R Session Information</em>. R package version 1.1.1. 2018. &lt;URL: <a href="https://github.com/r-lib/sessioninfo#readme" class="uri">https://github.com/r-lib/sessioninfo#readme</a>&gt;.</p>
<p>[4] dzhang32. <em>Detecting abberant splicing events from RNA-sequencing data</em>. <a href="https://github.com/dzhang32/dasper" class="uri">https://github.com/dzhang32/dasper</a> - R package version 0.99.0. 2020. DOI: 10.18129/B9.bioc.dasper. &lt;URL: <a href="http://www.bioconductor.org/packages/dasper" class="uri">http://www.bioconductor.org/packages/dasper</a>&gt;.</p>
<p>[5] A. Oleś, M. Morgan, and W. Huber. <em>BiocStyle: Standard styles for vignettes and other Bioconductor documents</em>. R package version 2.17.0. 2020. &lt;URL: <a href="https://github.com/Bioconductor/BiocStyle" class="uri">https://github.com/Bioconductor/BiocStyle</a>&gt;.</p>
<p>[6] R Core Team. <em>R: A Language and Environment for Statistical Computing</em>. R Foundation for Statistical Computing. Vienna, Austria, 2020. &lt;URL: <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>&gt;.</p>
<p>[7] H. Wickham. “testthat: Get Started with Testing”. In: <em>The R Journal</em> 3 (2011), pp. 5-10. &lt;URL: <a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="uri">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>&gt;.</p>
<p>[8] Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report Generation in R</em>. R package version 1.29. 2020. &lt;URL: <a href="https://yihui.org/knitr/" class="uri">https://yihui.org/knitr/</a>&gt;.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by David Zhang.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
